flink-clients/src/test/java/org/apache/flink/client/CliFrontendSavepointTest.java

```diff
@@ -178,65 +185,30 @@ public void testTriggerSavepointCustomTarget() throws Exception {
 	public void testDisposeSavepointSuccess() throws Exception {
 		replaceStdOutAndStdErr();
 
-		try {
-			String savepointPath = "expectedSavepointPath";
-			ActorGateway jobManager = mock(ActorGateway.class);
+		String savepointPath = "expectedSavepointPath";
 
-			Promise<Object> triggerResponse = new scala.concurrent.impl.Promise.DefaultPromise<>();
+		final CompletableFuture<Acknowledge> disposeCallFuture = new CompletableFuture<>();
 
-			when(jobManager.ask(
-					Mockito.eq(new DisposeSavepoint(savepointPath)),
-					any(FiniteDuration.class))).thenReturn(triggerResponse.future());
+		ClusterClient clusterClient = new DisposeSavepointClusterClient((String path, Time timeout) -> {
+			disposeCallFuture.complete(Acknowledge.get());
+			return CompletableFuture.completedFuture(Acknowledge.get());
+		});
 
-			triggerResponse.success(getDisposeSavepointSuccess());
+		try {
 
-			CliFrontend frontend = new MockCliFrontend(
-					CliFrontendTestUtils.getConfigDir(), jobManager);
+			CliFrontend frontend = new MockedCliFrontend(clusterClient);
 
 			String[] parameters = { "-d", savepointPath };
-			int returnCode = frontend.savepoint(parameters);
+			frontend.savepoint(parameters);
 
-			assertEquals(0, returnCode);
-			verify(jobManager, times(1)).ask(
-					Mockito.eq(new DisposeSavepoint(savepointPath)),
-					any(FiniteDuration.class));
+			disposeCallFuture.get();
 
 			String outMsg = buffer.toString();
 			assertTrue(outMsg.contains(savepointPath));
 			assertTrue(outMsg.contains("disposed"));
 		}
 		finally {
-			restoreStdOutAndStdErr();
-		}
-	}
-
-	/**
-	 * Tests that a disposal failure due a  ClassNotFoundException triggers a
-	 * note about the JAR option.
-	 */
-	@Test
-	public void testDisposeClassNotFoundException() throws Exception {
```

Comment:
The behaviour moved to `ClusterClient` but we are not testing it anymore.

Reply:
Yes indeed, will add it with the referenced commit.